### PR TITLE
event_manager_proxy: Repeat IPC transmission on failure

### PR DESCRIPTION
**Pull request by rakons**
This commit adds the functionality to repeat the transmission
on failure.

### CODE REVIEW

File: subsys/event_manager_proxy/event_manager_proxy.c [hunk 1]
```diff
@@ -288,7 +289,11 @@ static int send_event_to_remote(struct emp_ipc_data *ipc, const struct app_event
 	memcpy(buffer, eh, sizeof(buffer));
 	remote_eh->type_id = remote_ev;
 
-	int ret = ipc_service_send(&ipc->ept, buffer, sizeof(buffer));
+	for (size_t cnt = CONFIG_EVENT_MANAGER_PROXY_SEND_RETRIES; cnt > 0; --cnt) {
```
Comment (pdunaj) [inline subsys/event_manager_proxy/event_manager_proxy.c:292]: number of tries cannot be zero... `cnt = CONFIG_EVENT_MANAGER_PROXY_SEND_RETRIES + 1`?

**Review by pdunaj — CHANGES_REQUESTED** [review]

File: subsys/event_manager_proxy/event_manager_proxy.c (hunk 1 above)
Comment (rakons) [reply subsys/event_manager_proxy/event_manager_proxy.c]: Done

**Review by pdunaj — APPROVED** [review]

**Review by zycz — APPROVED** [review]

File: subsys/event_manager_proxy/event_manager_proxy.c [hunk 2]
```diff
@@ -288,7 +289,11 @@ static int send_event_to_remote(struct emp_ipc_data *ipc, const struct app_event
 	memcpy(buffer, eh, sizeof(buffer));
 	remote_eh->type_id = remote_ev;
 
-	int ret = ipc_service_send(&ipc->ept, buffer, sizeof(buffer));
+	for (size_t cnt = CONFIG_EVENT_MANAGER_PROXY_SEND_RETRIES + 1; cnt > 0; --cnt) {
+		ret = ipc_service_send(&ipc->ept, buffer, sizeof(buffer));
+		if (ret >= 0)
```
Comment (emob-nordic) [inline subsys/event_manager_proxy/event_manager_proxy.c:294]: Maybe checkpatch allows to do it but i don't!
I believe we were obliged to always use {} brackets, even for single line if statement.

**Review by emob-nordic — CHANGES_REQUESTED** [review]

File: subsys/event_manager_proxy/event_manager_proxy.c (hunk 2 above)
Comment (pdunaj) [reply subsys/event_manager_proxy/event_manager_proxy.c:294]: really? I don't think so...

Comment (emob-nordic) [reply subsys/event_manager_proxy/event_manager_proxy.c:294]: You are right, i heard it int the past, not even sure if its still applicable. Checkpatch doesn't complain so my nack is an abuse.

**Review by emob-nordic — APPROVED** [review]

File: subsys/event_manager_proxy/event_manager_proxy.c (hunk 2 above)
Comment (rakons) [reply subsys/event_manager_proxy/event_manager_proxy.c]: Fixed

**Comment by rakons** [comment]
Changes in the code provided in current release - no changelog entry required.

**Pull request merged**